tests/cid.rs

```diff
@@ -83,19 +83,30 @@ fn test_cid_not_as_bytes() {
     from_slice::<serde_bytes::ByteBuf>(&cbor_cid[2..])
         .expect("should have parsed an untagged CID as a byte array");
 
-    #[cfg(feature = "no-cid-as-bytes")]
-    {
-        #[derive(Debug, Deserialize)]
-        struct NewType(ByteBuf);
+    #[derive(Debug, Deserialize, PartialEq)]
+    struct NewType(ByteBuf);
 
-        #[derive(Debug, Deserialize)]
-        #[serde(untagged)]
-        enum BytesInEnum {
-            MyCid(NewType),
-        }
+    #[derive(Debug, Deserialize, PartialEq)]
+    #[serde(untagged)]
+    enum BytesInEnum {
+        MyCid(NewType),
+    }
+
+    // With the `no-cid-as-bytes` feature enabled, we make sure that it will error, when we try to
+    // decode a CID as bytes.
+    #[cfg(feature = "no-cid-as-bytes")]
+    from_slice::<BytesInEnum>(&cbor_cid)
+        .expect_err("shouldn't have parsed a tagged CID as byte array");
 
-        from_slice::<BytesInEnum>(&cbor_cid)
-            .expect_err("shouldn't have parsed a tagged CID as byte array");
+    // With that feature disabled, then it will decode the CID (without the TAG and the zero
+    // prefix) as bytes.
+    #[cfg(not(feature = "no-cid-as-bytes"))]
+    {
+        let cid_without_tag = &cbor_cid[5..];
+        assert_eq!(
+            from_slice::<BytesInEnum>(&cbor_cid).unwrap(),
+            BytesInEnum::MyCid(NewType(ByteBuf::from(cid_without_tag)))
```

Comment:
Cool, this is what I was looking for, complete loss of information. I think I can extrapolate from this to situations where there is a choice between types and it picks the wrong one because the metadata is gone.